### PR TITLE
Enhancements for support function keys in `editorManager`

### PR DESCRIPTION
**Pull request by swistach**
### Context
In #7838 we got a request to add support for <kbd>CTRL</kbd>/<kbd>SHIFT</kbd> + <kbd>INSERT</kbd>. However, these shortcuts should invoke native copy and paste actions.
Investigating the roots of problems, I noticed, we do not support function keys as we should.
The following table presents differences between Handsontable v8.4.0, Google Spreadsheets, Excel and OpenOffice:

| Key                                          | Handsontable                | GoogleSpreadsheet | Excel      | OpenOffice |
|----------------------------------------------|-----------------------------|-------------------|------------|------------|
| <kbd>AudioVolumeDown</kbd>                              | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>AudioVolumeMute</kbd>                              | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>AudioVolumeUp</kbd>                                | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>F13</kbd>                                          | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>F14</kbd>                                          | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>F15</kbd>                                          | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>F16</kbd>                                          | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>F17</kbd>                                          | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>F18</kbd>                                          | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>F19</kbd>                                          | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>Insert</kbd>                                       | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>MediaPlayPause</kbd>                               | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>MediaTrackNext</kbd>                               | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>MediaTrackPrevious</kbd>                           | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>MediaStop</kbd>                                    | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>NULL</kbd> (non-printable key without the keyCode) | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>NumLock</kbd>                                      | opens editor, clears textarea | does nothing        | does nothing | does nothing |
| <kbd>ScrollLock</kbd>                                   | opens editor, clears textarea | does nothing        | does nothing | does nothing |

This PR:
- adds missing keys to the `unicode`'s helper,
- renames helper's function (`isMetaKey` => `isFunctionKey`) - it's internal helper and it should not cause `breaking change`,
- adjusts `editorManager`'s behaviour to expected for non-printable keys,
- adds E2E tests for `editorManager` (with the default editor) for added function keys.

### How has this been tested?
A. Run test scripts.
B. Check the above list of keys in browsers manually.

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. #7838

### Affected project(s):
- [x] `handsontable`
- [x] `@handsontable/angular`
- [x] `@handsontable/react`
- [x] `@handsontable/vue`

### Checklist:
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: wrappers/react/test/_helpers.tsx [hunk 1]
```diff
@@ -32,12 +32,30 @@ export function mockElementDimensions(element, width, height) {
...
+  // const init = (event as any).initKeyboardEvent !== void 0 ? 'initKeyboardEvent' : 'initKeyEvent';
 
-  event[init](type, true, true, window, false, false, false, false, keyCode, 0);
+  // Chromium Hack
+  Object.defineProperty(newEvent, 'keyCode', {
+    get : function() {
+        return this.keyCodeVal;
+    }
+  });     
+  Object.defineProperty(newEvent, 'which', {
```
Comment (wojciechczerniak) [inline wrappers/react/test/_helpers.tsx:44]: Both `keyCode` and `which` are deprecated. Shouldn't we rather remove them from our code?

File: src/helpers/unicode.js [hunk 2]
```diff
@@ -40,18 +37,89 @@ export const KEY_CODES = {
...
+  PERIOD: 190,
   S: 83,
+  SCROLL_LOCK: 145,
+  SHIFT: 16,
+  SPACE: 32,
+  TAB: 9,
   V: 86,
   X: 88,
+  Y: 89,
+  Z: 90,
```
Comment (budnix) [inline src/helpers/unicode.js:70]: Generally, it's a great idea to sort the props alphabetically. But I feel that in that case, it is harder to read. Especially with the single letters in between :D WDYT about grouping those letters below. Something like:
```js
  SHIFT: 16,
  SPACE: 32,
  TAB: 9,
  A: 65,
  C: 67,
  D: 68,
  F: 70,
  L: 76,
  O: 79,
  P: 80,
  S: 83,
  V: 86,
  X: 88,
  Y: 89,
  Z: 90,
```

**Review by budnix — CHANGES_REQUESTED** [review]
I tested on macOS and Windows and didn't found any issues. I left some comments to discuss.

File: wrappers/react/test/_helpers.tsx (hunk 1 above)
Comment (swistach) [reply wrappers/react/test/_helpers.tsx:44]: Unfortunately, we use `keyCode` internally in many places:
- editorManager,
- walkontable.overlays,
- autocompleteEditor,
- dateEditor,
- handsontableEditor,
- selectEditor,
- textEditor, ​
- contextMenu.menu,
- filters (component/ui),
- mergeCells,
- undoRedo,
- checkboxRenderer,

I adjusted creating keyboard events accordingly to their specification (some differences in arguments order etc.). Initially, I was hopeful that we could use `new KeyboardEvent()`- however - it's not supported by IE (I assume we run wrappers' tests on IE).

File: src/helpers/unicode.js (hunk 2 above)
Comment (swistach) [reply src/helpers/unicode.js:70]: Yes, I agree. It is something that might improve readability.
Alternatively, I've been thinking about sorting keys by keyCode.
Thanks, @budnix 💪!

**Review by budnix — APPROVED** [review]
Thanks 💪

File: src/__tests__/editorManager.spec.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          keyDownUp(key.toLowerCase());
+
+          expect(activeEditor.isOpened()).toBe(true);
+        });
+      });
+    });
+
+    describe('should not begin editing', () => {
+      [
+        'ALT',
```
Comment (aninde) [inline src/__tests__/editorManager.spec.js:37]: I wonder if we should have a division into <kbd>Alt</kbd> and <kbd>AltGr</kbd> for better management of it in the future. https://en.wikipedia.org/wiki/AltGr_key. Each key works differently in a Handsontable.
<kbd>AltGr</kbd> (right alt key) with selection enables multiselection, left alt not.

**Review by aninde — COMMENTED** [review]
Works on Windows 10 and macOS 11.2.3 as expected. I propose the division of alt keys.

**Comment by aninde** [comment]
This information should be included in the documentation.

File: src/__tests__/editorManager.spec.js (hunk 3 above)
Comment (swistach) [reply src/__tests__/editorManager.spec.js:37]: <kbd>AltGr</kdb> under the hood uses <kbd>Ctrl</kdb>+<kbd>Alt</kdb>. We support selecting non-contiguous cells if user press <kbd>Ctrl</kbd> key.

From the `event.keyCode` (the property we use to recognize what key is used), we get `18` for both, `Alt` and `AltGraph`.

**Comment by swistach** [comment]
> This information should be included in the documentation.

I put that point on the checklist in the issue's description.

File: src/__tests__/editorManager.spec.js (hunk 3 above)
Comment (aninde) [reply src/__tests__/editorManager.spec.js:37]: > So I suggest moving that point to the separated issue due to we do not support `AltGr` anywhere in Handsontable.

Good point.

**Review by aninde — APPROVED** [review]
👍

File: docs/next/guides/accessories-and-menus/keyboard-navigation.md [hunk 4]
```diff
@@ -53,10 +53,10 @@ tags:
 | <kbd>Backspace</kbd> | <kbd>Backspace</kbd> | Empty cell |
 | <kbd>Delete</kbd>  | <kbd>Delete</kbd> | Empty cell |
 | <kbd>Shift</kbd> + <kbd>Enter ⏎</kbd> | <kbd>Shift</kbd> + <kbd>Enter ⏎</kbd> | Opens editor and move to the cell above current active cell (if exists) |
-| <kbd>Ctrl</kbd> + <kbd>Enter ⏎</kbd> | <kbd>Ctrl</kbd> + <kbd>Enter ⏎</kbd>  | Opens editor and adds new line in the cell |
-| <kbd>Ctrl</kbd> + <kbd>C</kbd>  | <kbd>Cmd</kbd> + <kbd>C</kbd> | Copy cell's content |
+| <kbd>Ctrl</kbd> + <kbd>Enter ⏎</kbd> | <kbd>Ctrl</kbd> + <kbd>Enter ⏎</kbd> | Opens editor and adds new line in the cell |
+| <kbd>Ctrl</kbd> + <kbd>C</kbd><br>or<br><kbd>Ctrl</kbd> + <kbd>Insert</kbd> | <kbd>Cmd</kbd> + <kbd>C</kbd><br>or<br><kbd>Ctrl</kbd> + <kbd>Insert</kbd> | Copy cell's content |
```
Comment (aninde) [inline docs/next/guides/accessories-and-menus/keyboard-navigation.md:57]: There is no insert button on `original` macOS keyboards. I can see on the Stack Overflow that the `Insert` shortcuts differ depending on the macOS version, on some the is <kbd>Fn</kbd> + <kbd>Enter</kbd>, on other <kbd>Shift</kbd> + <kbd>Fn</kbd> + <kbd>Enter</kbd>. 

On my macOS, there is no working shortcut that can simulate it. In order not to mix users, I suggest adding information that the `Insert` on the macOS should be mapped, or let's delete the info in macOS at all.

**Review by aninde — CHANGES_REQUESTED** [review]

**Review by budnix — DISMISSED** [review]

File: docs/next/guides/accessories-and-menus/keyboard-navigation.md (hunk 4 above)
Comment (swistach) [reply docs/next/guides/accessories-and-menus/keyboard-navigation.md]: I removed unnecessary shortcuts.

**Review by budnix — APPROVED** [review]

**Review by aninde — APPROVED** [review]
Great 🥇

**Pull request merged**